osu.Game.Modes.Osu/Objects/SliderCurve.cs

```diff
@@ -172,6 +172,9 @@ private Vector2 interpolateVertices(int i, double d)
         /// <param name="p1">End progress. Ranges from 0 (beginning of the slider) to 1 (end of the slider).</param>
         public void GetPathToProgress(List<Vector2> path, double p0, double p1)
         {
+            if (calculatedPath.Count == 0 && ControlPoints.Count > 0)
+                Calculate();

```

Comment:
Lazy calculation has the pretty significant downside that stuff may (potentially) stutter during gameplay rather than load at the beginning of the map.